Update field naming convention

BenjaminMichaelis opened the pull request:
## Summary

- Updated field names to follow coding guidelines.
- Put `_date` next to its wrapping property.
- Placed public field below the public property as it shouldn't generally be used.

Fixes #22822

BenjaminMichaelis committed:
Update field naming convention

- Updated field names to follow coding guidelines.
- Put `_date` next to its wrapping property.
- Placed public field below the public property as it shouldn't generally be used.

Fix #22822

Youssef1313 reviewed (changes requested):
Please update <https://github.com/BenjaminMichaelis/docs/blob/BenjaminMichaelis-Issue%2322822/docs/csharp/programming-guide/classes-and-structs/fields.md> as well. Thanks!

BillWagner commented on docs/csharp/programming-guide/classes-and-structs/snippets/fields/Program.cs:
```diff
@@ -3,25 +3,23 @@
-    // private field
-    private DateTime date;
 
-    // public field (Generally not recommended.)
```
This line should stay in the source. While not recommended, it is allowed. The comment does indicate that it's not the recommended practice.

You could / should change the name to "Day" to follow the conventions.

BenjaminMichaelis replied:
This public field has already been renamed to "Day" but I moved it below the property so it still exists. I moved it away from the top because I thought that it would be poor to put an example that is not recommended up top, as well as then it allows the private field to be by its wrapping property which I think could be good practice, and if that happens, the public property would have to be the first example seen if it is still up top, but I can add it back to the top again, and move the private field above the public field like originally. Does that sounds good to move it back to the original format ordering or keep it as is? I will update the .md as well.

BillWagner replied:
Thanks for clarifying @BenjaminMichaelis.  I missed it looking at the diffs. The declaration can stay where it is.

BillWagner reviewed (changes requested):
Thanks for putting this together @BenjaminMichaelis 

I had one change in the code for you to make. Also, as @Youssef1313 mentioned, before I merge this, you need to update the corresponding text to match the new variable names.

You can add commits to this branch, and that updates the PR.

BenjaminMichaelis committed:
Update fields.md to match Program.cs changes.

- Change `day` to `Day` to match coding standard casing as shown in Program.cs
- Update `.fieldname` to `._fieldName` to make field coding guidelines of prefix and camelCase clear.

BillWagner commented:
Thanks for making the  additional updates @BenjaminMichaelis 

I'll :shipit: now.

You should see the changes live on our site in the next day or two, on our regular publishing cycle.
